plugins/org.eclipse.elk.core.service/src/org/eclipse/elk/core/service/graphvisitors/CompoundGraphElementVisitor.java

```diff
@@ -0,0 +1,66 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Kiel University and others.
+ * 
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0  *******************************************************************************/
+package org.eclipse.elk.core.service.graphvisitors;
+
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.eclipse.elk.core.util.ElkUtil;
+import org.eclipse.elk.core.util.IGraphElementVisitor;
+import org.eclipse.elk.graph.ElkGraphElement;
+import org.eclipse.elk.graph.ElkNode;
+
+/**
+ * Class to apply many graph element visitors on the same graph.
+ */
+public class CompoundGraphElementVisitor implements IGraphElementVisitor {
+    
+    /**
+     * List of graph visitors which will be applied in the given order.
+     */
+    private List<IGraphElementVisitor> graphVisitors = new LinkedList<>();
```

Comment:
Any special reason for a `LinkedList`? We used them all over the place, but `ArrayList`s are now preferred.

Reply:
I guess ArrayList could work too.